pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix, MatrixUndefinedError
+
+
+def sum(*summands, do_queue=True, id=None):
+    """Top level sum function to create an instance of Sum"""
+    return Sum(*summands, do_queue=do_queue, id=id)
+
+
+def _sum(mats_gen, dtype=None, cast_like=None):
+    """Private sum method given a series of matrices of correct size.
+    Super inefficient method just as a proof of concept."""
+
+    res = None
+    try:
+        for i, mat in enumerate(mats_gen):
+            res = mat if i == 0 else math.add(res, mat)
+    except MatrixUndefinedError as error:
+        print(f"\nThe matrix method must be defined for all summands: \n")
+        raise error
+
+    if dtype is not None:
+        res = math.cast(res, dtype)
+    if cast_like is not None:
+        res = math.cast_like(res, cast_like)
+
+    return res
+
+
+class Sum(Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
+
+        combined_wires = qml.wires.Wires.all_wires([s.wires for s in summands])
+        combined_params = []
+        for s in summands:
+            combined_params += s.parameters
```

Comment:
```suggestion
        combined_params = [s.parameters for s in summands]
```
A little suggestion for conciseness.